ports/directx-dxc/directx-dxc-config.cmake.in

```diff
@@ -1,44 +1,24 @@
-
 get_filename_component(_dxc_root "${CMAKE_CURRENT_LIST_DIR}" PATH)
 get_filename_component(_dxc_root "${_dxc_root}" PATH)
 
-set(_dxc_exe "${_dxc_root}/@tool_path@")
-if (EXISTS "${_dxc_exe}")
-
-   add_library(Microsoft::DirectXShaderCompiler SHARED IMPORTED)
-   set_target_properties(Microsoft::DirectXShaderCompiler PROPERTIES
-      IMPORTED_LOCATION_RELEASE            "${_dxc_root}/@dll_dir@/@dll_name_dxc@"
-      IMPORTED_LOCATION_DEBUG              "${_dxc_root}/@dll_dir@/@dll_name_dxc@"
-      IMPORTED_IMPLIB_RELEASE              "${_dxc_root}/lib/@lib_name@"
-      IMPORTED_IMPLIB_DEBUG                "${_dxc_root}/lib/@lib_name@"
-      IMPORTED_SONAME_RELEASE              "@lib_name@"
-      IMPORTED_SONAME_DEBUG                "@lib_name@"
-      INTERFACE_INCLUDE_DIRECTORIES        "${_dxc_root}/include/directx-dxc"
-      IMPORTED_CONFIGURATIONS              "Debug;Release"
-      IMPORTED_LINK_INTERFACE_LANGUAGES    "C")
-
-   add_library(Microsoft::DXIL SHARED IMPORTED)
-   set_target_properties(Microsoft::DXIL PROPERTIES
-      IMPORTED_LOCATION_RELEASE            "${_dxc_root}/@dll_dir@/@dll_name_dxil@"
-      IMPORTED_LOCATION_DEBUG              "${_dxc_root}/@dll_dir@/@dll_name_dxil@"
-      IMPORTED_IMPLIB_RELEASE              "${_dxc_root}/lib/@lib_name@"
-      IMPORTED_IMPLIB_DEBUG                "${_dxc_root}/lib/@lib_name@"
-      IMPORTED_SONAME_RELEASE              "@lib_name@"
-      IMPORTED_SONAME_DEBUG                "@lib_name@"
-      INTERFACE_INCLUDE_DIRECTORIES        "${_dxc_root}/include/directx-dxc"
-      IMPORTED_CONFIGURATIONS              "Debug;Release"
-      IMPORTED_LINK_INTERFACE_LANGUAGES    "C")
-
-   target_link_libraries(Microsoft::DirectXShaderCompiler INTERFACE Microsoft::DXIL)
-
-   set(directx-dxc_FOUND TRUE)
-   set(DIRECTX_DXC_TOOL ${_dxc_exe})
-
-else()
-
-    set(directx-dxc_FOUND FALSE)
-
-endif()
+set(DIRECTX_DXC_TOOL "${_dxc_root}/@tool_path@" CACHE PATH "Location of the dxc tool")
+mark_as_advanced(DIRECTX_DXC_TOOL)
+
+add_library(Microsoft::DirectXShaderCompiler SHARED IMPORTED)
+set_target_properties(Microsoft::DirectXShaderCompiler PROPERTIES
+   IMPORTED_LOCATION                    "${_dxc_root}/@dll_dir@/@dll_name_dxc@"
+   IMPORTED_IMPLIB                      "${_dxc_root}/lib/@lib_name@"
```

Comment:
When I try to build for `RelWithDebInfo` it fails. Any idea why?

Reply:
I have to use `CMAKE_MAP_IMPORTED_CONFIG_RELWITHDEBINFO` for other ports to get Release to map to RelWithDebInfo.

The problem is that the new target doesn't work anymore with that scenario.  It looks like the fix is to add:

```
   MAP_IMPORTED_CONFIG_MINSIZEREL       ""
   MAP_IMPORTED_CONFIG_RELWITHDEBINFO   ""
```

Reply:
https://github.com/microsoft/vcpkg/pull/37485#issuecomment-2001789218